src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -1108,284 +1108,292 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
+
+                                                IsIncomingReplication = false;
+                                                _replicationInfo.ConflictManager.HandleConflictForDocument(context, doc.Id, doc.Collection, doc.LastModifiedTicks,
+                                                    document, rcvdChangeVector, doc.Flags);
+                                                continue;
+                                            }
 
-                                        // cluster tx has precedence over regular tx
+                                            // cluster tx has precedence over regular tx
 
-                                        if (hasLocalClusterTx)
-                                            goto case ConflictStatus.AlreadyMerged;
+                                            if (hasLocalClusterTx)
+                                                goto case ConflictStatus.AlreadyMerged;
 
-                                        if (hasRemoteClusterTx)
-                                            goto case ConflictStatus.Update;
+                                            if (hasRemoteClusterTx)
+                                                goto case ConflictStatus.Update;
 
-                                        break;
+                                            break;
 
-                                    case ConflictStatus.AlreadyMerged:
-                                        // we have to do nothing here
-                                        break;
+                                        case ConflictStatus.AlreadyMerged:
+                                            // we have to do nothing here
+                                            break;
 
-                                    default:
-                                        throw new ArgumentOutOfRangeException(nameof(conflictStatus),
-                                            "Invalid ConflictStatus: " + conflictStatus);
-                                }
+                                        default:
+                                            throw new ArgumentOutOfRangeException(nameof(conflictStatus),
+                                                "Invalid ConflictStatus: " + conflictStatus);
+                                    }
 
-                                break;
+                                    break;
 
-                            default:
-                                throw new ArgumentOutOfRangeException(item.GetType().ToString());
+                                default:
+                                    throw new ArgumentOutOfRangeException(item.GetType().ToString());
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            var errorMessage = BuildErrorMessage(item);
+                            throw new InvalidOperationException(errorMessage, e);
```

Comment:
I don't like that we change the type of the exception here... maybe add this to the `Data`?

Reply:
But then we will not see it in the logs.
I can just log it but then we will see this error twice in the logs (not so bad).
Or I can implement the logs to also write the data.

Reply:
1. We need to see that in the logs
2. We need to see that in the notification center as well